### PR TITLE
[GTK] Disable DMABuf renderer for NVIDIA proprietary drivers

### PR DESCRIPTION
**Pull request by carlosgcampos**
#### 745633e606ca70bf926545149beed35aab6450b7
<pre>
[GTK] Disable DMABuf renderer for NVIDIA proprietary drivers
<a href="https://bugs.webkit.org/show_bug.cgi?id=262607">https://bugs.webkit.org/show_bug.cgi?id=262607</a>

Reviewed by NOBODY (OOPS!).

Some NVIDIA users are reporting nothing is rendered, so let&apos;s just
disable the DMA-BUF renderer until we figure out how to make it work.

* Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:
(WebKit::isNVIDIA):
(WebKit::AcceleratedBackingStoreDMABuf::rendererBufferMode):
</pre>
https://github.com/WebKit/WebKit/commit/745633e606ca70bf926545149beed35aab6450b7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/20840 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/21247 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/21907 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/22729 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/19420 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/24486 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/21423 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/20715 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/21061 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/20833 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/18096 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/23583 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/17997 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/25208 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/19077 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/19091 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/23117 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/19671 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/16709 "14 flakes 2 failures") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/18913 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/23242 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/19487 "Built successfully") | | |

### CODE REVIEW

File: Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp [hunk 1]
```diff
@@ -55,6 +57,29 @@ static constexpr uint64_t s_dmabufInvalidModifier = ((1ULL << 56) - 1);
...
+    const char* disableGBM = getenv("WEBKIT_DMABUF_RENDERER_DISABLE_GBM");
+    if (!disableGBM || !strcmp(disableGBM, "0")) {
+        if (auto* device = WebCore::PlatformDisplay::sharedDisplay().gbmDevice())
+            platformDisplay = WebCore::PlatformDisplayGBM::create(device);
+    }
+#endif
+    if (!platformDisplay)
+        platformDisplay = WebCore::PlatformDisplaySurfaceless::create();
+
+    WebCore::GLContext::ScopedGLContext glContext(WebCore::GLContext::createOffscreen(platformDisplay ? *platformDisplay : WebCore::PlatformDisplay::sharedDisplay()));
```
Comment (mcatanzaro) [inline Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:77]: It looks like `WebCore::PlatformDisplaySurfaceless::create` never fails, so it should be impossible to ever use the shared display here, right?

**Review by mcatanzaro — APPROVED** [review]

File: Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp (hunk 1 above)
Comment (carlosgcampos) [reply Source/WebKit/UIProcess/gtk/AcceleratedBackingStoreDMABuf.cpp:77]: You are right

**Comment by bertogg** [comment]
> I cherry picked this in Debian (2.42.1-2) but the user says that this is still failing: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1052055#30

This seems to work when the `libnvidia-egl-gbm1` package is installed.

**Comment by mcatanzaro** [comment]
So, what do we want to do here?

**Comment by sergiomb2** [comment]
> So, what do we want to do here?

I think we should apply it on Fedora , since September I have this bug `export WEBKIT_DISABLE_DMABUF_RENDERER=1` make me work geany markdown plugin , biglybt which user eclipse-swt  , and eclipse web browsers

**Comment by mcatanzaro** [comment]
> I think we should apply it on Fedora , since September I have this bug export WEBKIT_DISABLE_DMABUF_RENDERER=1 make work for me, markdown plugin of geany, web browsers of biglybt which use eclipse-swt and eclipse web browsers

Is there a bug report for the problem you're encountering?

This was intended as a workaround for [this bug](https://bugs.webkit.org/show_bug.cgi?id=261874) that is already fixed.

**Comment by sergiomb2** [comment]
> > I think we should apply it on Fedora , since September I have this bug export WEBKIT_DISABLE_DMABUF_RENDERER=1 make work for me, markdown plugin of geany, web browsers of biglybt which use eclipse-swt and eclipse web browsers
> 
> Is there a bug report for the problem you're encountering?
> 
> This was intended as a workaround for [this bug](https://bugs.webkit.org/show_bug.cgi?id=261874) that is already fixed.

in fedora bugzilla, no , but I can open one , I'm using Fedora 39 and this is not fixed .

I just found the solution after saw the message "AcceleratedSurfaceDMABuf was unable to construct a complete framebuffer" and read https://gitlab.archlinux.org/archlinux/packaging/packages/webkit2gtk/-/issues/1

**Comment by mcatanzaro** [comment]
> in fedora bugzilla, no , but I can open one , I'm using Fedora 39 and this is not fixed .

No, a bug report in Red Hat Bugzilla will not help.

The dmabuf feature was released over 6 months ago, in 2.42. We were considering disabling it due to users reporting that it was broken on NVIDIA graphics, but all bug reports have since been fixed, and it's been 6 months now since the feature rolled out, which is a pretty long time. I think you need to report a new bug on WebKit Bugzilla (not Red Hat Bugzilla), WebKitGTK component for the problem you are encountering, so it can be evaluated.

I'm going to close this for now.

**Comment by sergiomb2** [comment]
ah maybe is just Fedora 39 that haven't the latest releases , just after upgrade my box from f38 to f39 I found this problem but I had already the problem on F38 ...

**Comment by sergiomb2** [comment]
After all, I was mistaken webkit2gtk4.0  and webkit2gtk4.1 are updated [1] but "AcceleratedSurfaceDMABuf was unable to construct a complete framebuffer" still appears and WEBKIT_DISABLE_DMABUF_RENDERER=1 still need to be used .
Should I open a bug report on bugzilla ?

[1]
webkit2gtk4.0                                   x86_64      2.44.0-2.fc39 
webkit2gtk4.1                                   x86_64      2.44.0-2.fc39

**Comment by mcatanzaro** [comment]
> Should I open a bug report on bugzilla ?

Yes. Use WebKit Bugzilla, WebKitGTK component.

**Pull request closed without merging**